agent/agent.go

```diff
@@ -119,14 +117,28 @@ func (a *Agent) Listen(ipString string, port int) error {
 	if err != nil {
 		return err
 	}
+	a.Logger.Info("agent successfully subscribed", zap.String("topic", topicName))
 
-	readLoop(sub, host.ID(), a.Config.MsgReceivedFn)
+	// read messages and print stats to logger
+	go func() {
+		for {
+			raw, err := sub.Next(context.Background())
+			if err != nil {
+				fmt.Printf("Peer %v error receiving message on topic: %v\n", host.ID(), err)
+				continue
+			}
+			a.Logger.Info("message received",
```

Comment:
We should log message too or at least some part of message which is unique for each message(full message logging will make log file too big) in order to make statistics later.